pkg/intermediate/aggregate.go

```diff
@@ -0,0 +1,284 @@
+package intermediate
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/registry"
+)
+
+type aggregation struct {
+	// tupleRecordMap maps each connection with its records
+	tupleRecordMap map[Tuple][]entities.Record
+	// tupleRecordLock allows multiple readers or one writer at the same time
+	tupleRecordLock sync.RWMutex
+	// workerPool is for storing worker channels to process the messages
+	workerPool chan chan *entities.Message
+	// messageChan is the channel to receive the message
+	messageChan chan *entities.Message
+	// workerNum is the number of workers to process the messages
+	workerNum int
+	// workerList is the list of workers
+	workerList []*worker
+	// correlateFields are the fields to be filled in correlating process
+	correlateFields []string
+}
+
+type Tuple struct {
```

Comment:
Should we name this struct something like ConnectionKey or FlowKey?
Tuple may sound too general for a public type?

Reply:
There is no tuples type in golang, so it might be ok.
I do agree the name `FlowKey` is more informative. 
We use `ConnectionKey` in Antrea, however, there is no context of connection here, so `FlowKey` is good.